.github/workflows/lint.yml

```diff
@@ -0,0 +1,25 @@
+name: Lint
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone repo
+        uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # v4.2.0
+
+      - name: Set up python
+        uses: actions/setup-python@f677139bbe7f9c59b41e40162b753c062f5d49a3 # 5.2.0
+        with:
+          python-version: 3.8
+
+      - name: Install dependencies
+        run: pip install black nbdev==2.3.25 pre-commit
+
+      - name: Run pre-commit
+        run: pre-commit run --files hierarchicalforecast/*
```

Comment:
Oh dear, did we not have any pre-commit hooks? 🤯

Reply:
Nope, I was giving the contributing instructions in the other PR and realized that we don't even support what I stated there, haha....